crates/jstzd/src/task/octez_client.rs

```diff
@@ -0,0 +1,156 @@
+use super::{endpoint::Endpoint, octez_node::OctezNode};
+use std::path::PathBuf;
+use tempfile::{tempdir, TempDir};
+
+#[derive(Default)]
+pub struct OctezClientBuilder {
+    binary_path: Option<PathBuf>,
+    // if None, use temp directory
+    base_dir: Option<PathBuf>,
+    // if None, use localhost:8732 (the default endpoint for octez-node)
+    endpoint: Option<OctezNodeEndpoint>,
+    disable_unsafe_disclaimer: bool,
+}
+
+impl OctezClientBuilder {
+    pub fn new() -> Self {
+        OctezClientBuilder::default()
+    }
+    pub fn set_path(mut self, path: PathBuf) -> Self {
+        self.binary_path = Some(path);
+        self
+    }
+    pub fn set_base_dir(mut self, base_dir: PathBuf) -> Self {
+        self.base_dir = Some(base_dir);
+        self
+    }
+    pub fn set_endpoint(mut self, endpoint: OctezNodeEndpoint) -> Self {
+        self.endpoint = Some(endpoint);
+        self
+    }
+    pub fn set_disable_unsafe_disclaimer(
+        mut self,
+        disable_unsafe_disclaimer: bool,
+    ) -> Self {
+        self.disable_unsafe_disclaimer = disable_unsafe_disclaimer;
+        self
+    }
+    pub fn build(mut self) -> OctezClient {
+        let base_dir = self
+            .base_dir
+            .map_or(OctezClientDirectory::default(), OctezClientDirectory::Path);
+        OctezClient {
+            binary_path: self.binary_path.take(),
+            base_dir,
+            endpoint: self.endpoint.unwrap_or_default(),
+            disable_unsafe_disclaimer: self.disable_unsafe_disclaimer,
+        }
+    }
+}
+
+pub enum OctezNodeEndpoint {
+    RpcEndpoint(Endpoint),
+    Node(OctezNode),
+}
+
+impl Default for OctezNodeEndpoint {
+    fn default() -> Self {
+        OctezNodeEndpoint::RpcEndpoint(Endpoint::localhost(8732))
+    }
+}
+
+impl From<Endpoint> for OctezNodeEndpoint {
+    fn from(endpoint: Endpoint) -> Self {
+        OctezNodeEndpoint::RpcEndpoint(endpoint)
+    }
+}
+impl From<OctezNode> for OctezNodeEndpoint {
+    fn from(node: OctezNode) -> Self {
+        OctezNodeEndpoint::Node(node)
+    }
+}
+
+pub enum OctezClientDirectory {
```

Comment:
Should implement `impl TryFrom<Path> for Directory` that validates Path is a directory or something similar along those lines.

Reply:
i already added the validation logic as part of `OctezClientBuilder::build` where we validates the binary path and the directory. I think renaming it to `Directory` and adding validation for it would be too generic to be in this file and out of scope of this PR.
We can leave the implementation in the build for now and create a more generic "File" and "Directory" struct that validates the path in a later PR. That way we can make OctezNode and other Nodes use these structs too. wdyt?

Reply:
I would imagine that Directory could be re-used by other components so having default validation implemented for it would be ideal

Reply:
i think we could work on that once it's needed. probably better to define it outside of otctez_client in that case

Reply:
Hmm I'm not satisfied that you are able to create a `Directory` that isn't a directory. It leaks the semantic.

Secondly, the author who ultimately makes the change may not know about your code changes and miss refactoring your code.  And I agree that this should be outside `octez_client.rs`, perhaps in `mod.rs`, sorry I missed that.